hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -355,11 +356,19 @@ public static String createNewFileId(String idPfx, int id) {
     return String.format("%s-%d", idPfx, id);
   }
 
+  private static String extractCommonLogFileName(String fileName) {
+    if (fileName.endsWith(HoodieCDCUtils.CDC_LOGFILE_SUFFIX)) {
+      return fileName.substring(0, fileName.length() - HoodieCDCUtils.CDC_LOGFILE_SUFFIX.length());
+    }
+    return fileName;
```

Comment:
not a big fan of having a special handling of cdc filenames. can we fix regex alone to achieve it?

Reply:
then have to modify the origin pattern, like https://github.com/apache/hudi/pull/7128.